pym/bob/scm/git.py

```diff
@@ -106,6 +107,7 @@ def __init__(self, spec, overrides=[], secureSSL=None, stripUser=None):
         self.__references = spec.get('references')
         self.__dissociate = spec.get('dissociate', False)
         self.__resolvedReferences = []
+        self.__useBranchAndCommit = spec.get('useBranchAndCommit', useBranchAndCommit)
```

Comment:
Why would you ever want to override this on a per-SCM basis? I would only add this if there is a valid use case.

Reply:
AFAICR I put it into the spec to have it set to a non `None` value in `__checkoutTag`.

Reply:
No, I don't want this to be set able per SCM. But I only have access to the policy when the scm is constructed from the input.py Scm-class. When the inkoker calls `getScm` I don't have access to the policy - or am I overlooking something?

Reply:
> When the inkoker calls getScm I don't have access to the policy - or am I overlooking something?

No, but then there is no need to also look in the spec. So the following should be sufficient:

```suggestion
        self.__useBranchAndCommit = useBranchAndCommit
```

Reply:
No it's not. Adding a debug print after `self.__useBranchAndCommit = useBranchAndCommit` shows:

```
useBranchAndCommit: True
useBranchAndCommit: None
```

within a `bob dev..` run. In `__checkoutTag` it's always `None`.

Reply:
You're absolutely right. Sorry for the noise. :see_no_evil: I totally forgot that the Scm instances are serialized and de-serialized before being executed.